spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayProperties.java

```diff
@@ -133,7 +133,7 @@ public void setInitSqls(List<String> initSqls) {
 	}
 
 	public boolean isCreateDataSource() {
-		return this.url != null && this.user != null;
+		return this.url != null || this.user != null;
```

Comment:
Support for datasource with just user?

Reply:
Sure, we want to be able to support being able to set a flyway user without having to explicitly set a flyway url. We should fallback to the default datasource url when provided with just a flyway user.